src/linker/ref/Linker/Annotations.cs

```diff
@@ -2,17 +2,27 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Collections.Generic;
 using Mono.Cecil;
-using Mono.Cecil.Cil;
 
 namespace Mono.Linker {
 
 	public partial class AnnotationStore
 	{
 		internal AnnotationStore() {}
+
+		public List<OverrideInformation> GetOverrides (MethodDefinition method) { throw null; }
```

Comment:
Should we try to change these APIs to be a bit more restrictive - given that they are becoming an official public API.
For example this one could return `IEnumerable`. And the `GetCustomAnnotations` could be replaced by something like `TryGetCustomAnnotation(object key, IMetadataTokenProvider item, out object value)` and `SetCustomAnnotation(object key, IMetadataTokenProvider item, object value)`. Basically trying to hide the implementation details.

Reply:
That makes sense to me (will change that in follow up PR)